dts/arm/st/stm32f446-pinctrl.dtsi

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2017 Pushpal Sidhu
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <dt-bindings/pinctrl/stm32-pinctrl.h>
+
+/ {
+	soc {
+		pinctrl: pin-controller {
+			usart2_pins_b: usart2@1 {
+				rx_tx {
+					rx = <STM32_PIN_PD6 (STM32_PINMUX_ALT_FUNC_7 | STM32_PUPDR_NO_PULL)>;
+					tx = <STM32_PIN_PD5 (STM32_PINMUX_ALT_FUNC_7 | STM32_PUSHPULL_NOPULL)>;
+				};
+			};
+			usart3_pins_c: usart3@2 {
+				rx_tx {
+					rx = <STM32_PIN_PC11 (STM32_PINMUX_ALT_FUNC_7 | STM32_PUSHPULL_PULLUP)>;
+					tx = <STM32_PIN_PC10 (STM32_PINMUX_ALT_FUNC_7 | STM32_PUSHPULL_PULLUP)>;
+				};
+			};
+			usart6_pins_a: usart6@0 {
+				rx_tx {
+					rx = <STM32_PIN_PC7 (STM32_PINMUX_ALT_FUNC_8 | STM32_PUSHPULL_PULLUP)>;
+					tx = <STM32_PIN_PC6 (STM32_PINMUX_ALT_FUNC_8 | STM32_PUSHPULL_PULLUP)>;
+				};
+			};
+			usart6_pins_b: usart6@1 {
+				rx_tx {
+					rx = <STM32_PIN_PG9 (STM32_PINMUX_ALT_FUNC_8 | STM32_PUSHPULL_PULLUP)>;
+					tx = <STM32_PIN_PG14 (STM32_PINMUX_ALT_FUNC_8 | STM32_PUSHPULL_PULLUP)>;
+				};
+			};
+		};
+	};
+};
```

Comment:
Without any board to configure, I wonder if we should even provide any pin configuration here.
I guess this make the need for a board to support even more required.

I let @galak decide on what to do next with this PR

Reply:
In my opinion, I would say that yes, we should provide this. It's an architecture specific dts file; boards using it can override things as needed, but these are purely what come from the datasheet.

However, since the Zephyr project isn't fully using dts file as intended (e.g. uart pin control is done in the pinmux driver, usart in device tree), maybe not unless the plan it to move to full device tree going forward.

Reply:
These are not purely what come from datasheet as you are here proposing abitrary pinctrl choices (PG9/PG14 is one of possible rx/tx combinations and not the only solutions). Hence we'll have to maintain this combination while it is not backed up by any board that we could test.
Point is purely about maintenance: until we have a board to test & use it, this code is purely hypothetical. If a change is made, there is no way to check if it is right/wrong or safe/unsafe.

Reply:
Right, I totally agree. I don't think we should have *any* pinctrl type device tree file at the SoC level. Instead, we need to let the board layout guys decide for themselves what pins to use, not inherit some random pinout.

I've been sitting here thinking about why there are even pinctrl files at the SoC level. Do you have any insight / comment? Honestly, I think they should be fully stripped and moved into board specific files only. The SoC files need to only implement things such as the following:

E.g. an SoC with CAN: set the proper compat / reg / interrupts / clocks etc at the SoC file, board level inherits SoC file and adds pinctrl as needed. Take a look at the imx6 dts files in Linux. They did a fairely decent job at it.

Reply:
Point is, for now, most of the boards use about same "pinmuxes", having definitions there avoids duplication. So that's a kind of balance. This is also the way it is done today on STM32 dts in Linux.
So for now, we'll stick to that. Though agreed that moving all this to boards is an idea to consider for a latter step

Reply:
Ok, I'll at least push something that removes this usart2 node.